app/models/specification_wrapper.rb

```diff
@@ -48,6 +49,20 @@ def validation_errors
         results
       end
 
+      def validate_public_access
+        return validate_http if @specification.source[:http]
+        return validate_git if @specification.source[:git]
+        true
+      end
+
+      def validate_http
+        system("curl -IL #{ Shellwords.escape @specification.source[:http] } -f")
```

Comment:
Why not just use `validate_url` from Core? https://github.com/CocoaPods/Core/blob/816eda9f90dd8fc54901b0fd4b768691fc1295bf/lib/cocoapods-core/http.rb#L35-39